File: Baekjoon/Python/Implementation/Q13335.py
# 트럭 / 구현, 시뮬레이션
import sys
input = sys.stdin.readline
n, w, l = map(int, input().split())
trucks = list(map(int, input().split()))
q = []
time = 0

while len(trucks) > 0 or len(q) > 0:
    time += 1
    
    if q and time - q[0][1] == w:
        l += q.pop(0)[0]
    
    if trucks and trucks[0] <= l:
        truck = trucks.pop(0)
        l -= truck
        q.append([truck, time])

print(time)

File: Baekjoon/Swift/Implementation/Q13335.swift
//
//  Q13335.swift
//  BOJ_PS
//
//  Created by 김태훈 on 2021/06/02.
//  트럭 / 구현, 시뮬레이션

typealias Truck = (w: Int, t: Int)

var nwl = readLine()!.split(separator: " ").map { Int(String($0))! }
var n = nwl[0]
var w = nwl[1]
var l = nwl[2]

var trucks = readLine()!.split(separator: " ").map { Int(String($0))! }
var time = 0
var q = [Truck]()

while !trucks.isEmpty || !q.isEmpty {
    time += 1
    
    // 트럭이 다리에서 빠져나올 때
    if let t = q.first?.t, time - t == w {
        l += q.removeFirst().w
    }
    
    // 트럭이 다리에 진입할 수 있을 때
    if let truck = trucks.first, truck <= l {
        l -= trucks.removeFirst()
        q.append((truck, time))
    }
}

print(time)
